samza-core/src/main/java/org/apache/samza/operators/impl/OperatorImpl.java

```diff
@@ -102,43 +100,34 @@ public final void init(Config config, TaskContext context) {
       throw new IllegalStateException(String.format("Attempted to initialize Operator %s after it was closed.", opId));
     }
 
-    this.highResClock = createHighResClock(config);
+    this.highResClock = createHighResClock(context.getJobContext().getConfig());
     registeredOperators = new HashSet<>();
     prevOperators = new HashSet<>();
     inputStreams = new HashSet<>();
-    MetricsRegistry metricsRegistry = context.getMetricsRegistry();
+    // TODO SAMZA-1935: the objects that are only accessible through TaskContextImpl should be moved somewhere else
+    TaskContextImpl taskContext = (TaskContextImpl) context.getTaskContext();
+    MetricsRegistry metricsRegistry = taskContext.getTaskMetricsRegistry();
     this.numMessage = metricsRegistry.newCounter(METRICS_GROUP, opId + "-messages");
     this.handleMessageNs = metricsRegistry.newTimer(METRICS_GROUP, opId + "-handle-message-ns");
     this.handleTimerNs = metricsRegistry.newTimer(METRICS_GROUP, opId + "-handle-timer-ns");
-    this.taskName = context.getTaskName();
+    this.taskName = taskContext.getTaskModel().getTaskName();
 
-    TaskContextImpl taskContext = (TaskContextImpl) context;
     this.eosStates = (EndOfStreamStates) taskContext.fetchObject(EndOfStreamStates.class.getName());
     this.watermarkStates = (WatermarkStates) taskContext.fetchObject(WatermarkStates.class.getName());
     this.controlMessageSender = new ControlMessageSender(taskContext.getStreamMetadataCache());
-
-    if (taskContext.getJobModel() != null) {
-      ContainerModel containerModel = taskContext.getJobModel().getContainers()
-          .get(context.getSamzaContainerContext().id);
-      this.taskModel = containerModel.getTasks().get(taskName);
-    } else {
-      this.taskModel = null;
-      this.usedInCurrentTask = true;
-    }
-
-    this.taskContext = taskContext;
-    handleInit(config, taskContext);
+    this.taskModel = taskContext.getTaskModel();
```

Comment:
Do you know what this was doing earlier with the job model null check? When would it be null? Can't tell from the PR, is the `usedInCurrentTask` logic still the same?